guidelines/design-guidelines.md

```diff
@@ -31,4 +45,156 @@ Example:
+
+An example where we failed in the past to apply this is the `Headers` static class. It contains everything and the kitchen sink when it comes to headers.
+
+
+## Namespace rules for NServiceBus core and downstream components
+
+### Principles
+
+* Namespaces are used to guide external users and to uniquely identify types.
+* Folders are used to structure code for our internal purposes and don't necessarily need to align with the namespaces.
+* Our public API has two main consumers: developers building business solutions and developers extending NServiceBus. 
+
+### Core rules 
+
+* For public types relevant to business developers, we use the `NServiceBus` namespace to make them more discoverable.
+* For internal types, we also use the `NServiceBus` namespace. This allows for unique identification of said types in logs and stack traces.
+* For public types targeted at extensibility developers, we use `NServiceBus.{ExtensibilityPoint}` to hide types irrelevant for business developers but still make them discoverable for developers extending NServiceBus. 
+   - Examples of currently used extensibility namespaces
+      * `NServiceBus.Transport`
+      * `NServiceBus.Persistence`
+      * `NServiceBus.Serialization`
+      * `NServiceBus.Logging`
+      
+### Rules for downstream components
+
+* For public types relevant to business developers, we use the `NServiceBus` namespace to make them more discoverable and to avoid extra using statements when the component is used.
+* For internal types, we use the root [`{Component}`](naming.md) namespace. This allows for unique identification of said types in logs and stack traces.
+   - Examples: `NServiceBus.Gateway`, `NServiceBus.Persistence.AzureStorage` etc.
+* For public types targeted at extensibility developers, we use the root [`{Component}`](naming.md) namespace to hide types irrelevant for business developers but still make them discoverable for developers extending NServiceBus. 
+
+
+## Component naming rules
+
+Components can be split up into two types: ones that are unique and one that belongs to a category.
+
+The ones that are unique should be named appropriately. E.g. `NServiceBus.Gateway`,  `NServiceBus.Callbacks` etc
+
+Components belonging to a category should be named `NServiceBus.{Category}.*`. E.g. `NServiceBus.Persistence.AzureStorage`, `NServiceBus.Persistence.MongoDb`
+
+
+### Current categories
+
+This is the current list and will be expanded as needed
+
+* `Persistence` - Adapters for persistence infrastructure 
+* `Transport` - Adapters for queuing infrastructure
```

Comment:
"transport" vs "queuing": should the word "transport" be somewhere in the clarifying sentence?